sdk/tables/Azure.Data.Tables/src/TableTransactionalBatch.cs

```diff
@@ -290,57 +141,104 @@ public virtual bool TryGetFailedEntityFromException(RequestFailedException excep
+            _requestMessages.Enqueue(entity.RowKey);
+            _requestLookup[entity.RowKey] = new BatchItem(requestType, entity, null);
         }
 
         /// <summary>
-        /// Builds an ordered list of <see cref="HttpMessage"/>s containing the batch sub-requests.
+        /// Gets the collection of entities represented in the batch.
         /// </summary>
-        /// <returns></returns>
-        private List<(ITableEntity Entity, HttpMessage HttpMessage)> BuildOrderedBatchRequests()
+        /// <exception cref="NotImplementedException"></exception>
+        public IReadOnlyList<ITableEntity> GetEntities()
         {
-            var orderedList = _requestMessages.ToList();
-            foreach (var item in orderedList)
-            {
-                _changeset.AddContent(new RequestRequestContent(item.HttpMessage.Request));
-            }
-            return orderedList;
+            return _requestMessages
+                .ToList()
+                .Select(rk => _requestLookup[rk].Entity)
+                .ToList()
+                .AsReadOnly();
         }
 
         /// <summary>
-        /// Creates a pipeline to use for processing sub-operations before they
-        /// are combined into a single multipart request.
+        /// Clears the contents of the batch operation.
         /// </summary>
-        /// <returns>A pipeline to use for processing sub-operations.</returns>
-        private static HttpPipeline CreateBatchPipeline()
+        public void ClearOperations()
         {
-            // Configure the options to use minimal policies
-            var options = new TableClientOptions();
-            options.Diagnostics.IsLoggingEnabled = false;
-            options.Diagnostics.IsTelemetryEnabled = false;
-            options.Diagnostics.IsDistributedTracingEnabled = false;
-            options.Retry.MaxRetries = 0;
+            _requestLookup.Clear();
+            _requestMessages = new ConcurrentQueue<string>();
+        }
 
-            // Use an empty transport so requests aren't sent
-            options.Transport = new MemoryTransport();
+        /// <summary>
+        /// Removes the specified entity from the batch
+        /// </summary>
+        /// <param name="entity"></param>
+        /// <returns></returns>
+        /// <exception cref="NotImplementedException"></exception>
+        public bool RemoveEntity(ITableEntity entity)
+            => RemoveEntities(new[] { entity });
 
-            // Use the same authentication mechanism
-            return HttpPipelineBuilder.Build(options);
+        /// <summary>
+        /// Removes the specified entity from the batch
+        /// </summary>
+        /// <param name="entities"></param>
+        /// <returns></returns>
+        /// <exception cref="NotImplementedException"></exception>
+        public bool RemoveEntities(IEnumerable<ITableEntity> entities)
```

Comment:
This name sounds too much like a service operation. DeleteEntity vs RemoveEntity is very subtle.

Reply:
`RemoveEntityOperations`?

Reply:
Yeah